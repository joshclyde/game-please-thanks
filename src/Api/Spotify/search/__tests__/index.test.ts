import axios from "axios";

import { spotifySearch } from "../";

jest.mock(`axios`, () => {
  return {
    get: jest.fn(),
  };
});

const mockedAxios = axios as jest.Mocked<typeof axios>;

describe(`Spotify Search`, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test(`should return the response from spotify search`, async () => {
    mockedAxios.get.mockReturnValueOnce(Promise.resolve(`this is the search result`));
    const result = await spotifySearch({
      accessToken: `ACCESSTOKENTESTVALUE`,
      q: `bts`,
      type: `album`,
      market: `from_token`,
      limit: 12,
    });
    expect(result).toMatchSnapshot();
  });
  test(`should call axios with proper values`, async () => {
    mockedAxios.get.mockReturnValueOnce(Promise.resolve(`this is the search result`));
    await spotifySearch({
      accessToken: `ACCESSTOKENTESTVALUE`,
      q: `bts`,
      type: `album`,
      market: `from_token`,
      limit: 12,
    });
    expect(mockedAxios.get.mock.calls).toMatchSnapshot();
  });
});
