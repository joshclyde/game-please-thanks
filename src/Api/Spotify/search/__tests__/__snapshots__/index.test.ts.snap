// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotify Search should call axios with proper values 1`] = `
Array [
  Array [
    "https://api.spotify.com/v1/search",
    Object {
      "headers": Object {
        "Authorization": "Bearer ACCESSTOKENTESTVALUE",
      },
      "params": Object {
        "include_external": undefined,
        "limit": 12,
        "market": "from_token",
        "offset": undefined,
        "q": "bts",
        "type": "album",
      },
    },
  ],
]
`;

exports[`Spotify Search should return the response from spotify search 1`] = `"this is the search result"`;
